### PR TITLE
feat: Add a new account roles data source

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
- add a new data source for account rolers to be consistent with resources
- prove that database role with the same prefix is not in the output of SHOW ROLES
- deprecate the `roles` data source (it will be removed in v1-ready branch)

## Test Plan

* [x] acceptance tests

* [ ] …

## References

https://docs.snowflake.com/en/sql-reference/sql/show-roles

### CODE REVIEW

File: docs/data-sources/account_roles.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+page_title: "snowflake_account_roles Data Source - terraform-provider-snowflake"
+subcategory: ""
+description: |-
+  Datasource used to get details of filtered account roles. Filtering is aligned with the current possibilities for SHOW ROLES https://docs.snowflake.com/en/sql-reference/sql/show-roles query (like and in_class are all supported). The results of SHOW are encapsulated in one output collection.
+---
+
+# snowflake_account_roles (Data Source)
+
+Datasource used to get details of filtered account roles. Filtering is aligned with the current possibilities for [SHOW ROLES](https://docs.snowflake.com/en/sql-reference/sql/show-roles) query (`like` and `in_class` are all supported). The results of SHOW are encapsulated in one output collection.
```
Comment (sfc-gh-jcieslak) [inline docs/data-sources/account_roles.md:10]: nit: `Data source`

File: pkg/datasources/account_roles.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var accountRolesSchema = map[string]*schema.Schema{
+	"like": {
```
Comment (sfc-gh-jcieslak) [inline pkg/datasources/account_roles.go:20]: nit: you can use predefined like schema

File: pkg/datasources/account_roles.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		Optional:    true,
+		Description: "Filters the output with **case-insensitive** pattern, with support for SQL wildcard characters (`%` and `_`).",
+	},
+	"in_class": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: resources.IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		Description:      "Filters the SHOW GRANTS output by class name.",
+	},
+	"roles": {
```
Comment (sfc-gh-jcieslak) [inline pkg/datasources/account_roles.go:31]: `roles` or `account_roles`?

File: pkg/datasources/account_roles.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+func ReadAccountRoles(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	req := sdk.NewShowRoleRequest()
+
+	if likePattern, ok := d.GetOk("like"); ok {
+		req.WithLike(sdk.NewLikeRequest(likePattern.(string)))
+	}
```
Comment (sfc-gh-jcieslak) [inline pkg/datasources/account_roles.go:65]: nit: `handleLike`

File: pkg/datasources/account_roles_acceptance_test.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	return func(state *terraform.State) error {
+		err := accountRolesDataSourceContainsRole(name, comment)(state)
+		if err != nil && err.Error() == fmt.Sprintf("role %s not found", name) {
+			return nil
+		}
+		return fmt.Errorf("expected %s not to be present", name)
+	}
+}
+
+func accountRolesDataSourceContainsRole(name string, comment string) func(s *terraform.State) error {
```
Comment (sfc-gh-jcieslak) [inline pkg/datasources/account_roles_acceptance_test.go:90]: nit: we could this mechanism later on in custom assertions to assert a given returned item. Currently, we always take the first item.

File: pkg/datasources/account_roles.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+					},
+				},
+			},
+		},
+	},
+}
+
+func AccountRoles() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: TrackingReadWrapper(datasources.Roles, ReadRoles),
```
Comment (sfc-gh-jcieslak) [inline pkg/datasources/account_roles.go:52]: shouldn't it use `datasources.AccountRoles`?

File: pkg/datasources/account_roles_acceptance_test.go (hunk 5 above)
Comment (sfc-gh-jmichalak) [reply pkg/datasources/account_roles_acceptance_test.go:90]: True, added a small note in the generator README.

File: pkg/datasources/account_roles.go (hunk 4 above)
Comment (sfc-gh-jmichalak) [reply pkg/datasources/account_roles.go]: Done.

File: pkg/datasources/account_roles.go (hunk 3 above)
Comment (sfc-gh-jmichalak) [reply pkg/datasources/account_roles.go]: Changed.

File: pkg/datasources/account_roles.go (hunk 2 above)
Comment (sfc-gh-jmichalak) [reply pkg/datasources/account_roles.go]: Done.

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-jcieslak — DISMISSED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**